src/containers/subjects-search/SubjectsSearch.jsx

```diff
@@ -0,0 +1,152 @@
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import SearchIcon from '@mui/icons-material/Search'
+import { Box, InputAdornment, TextField, Typography } from '@mui/material'
+import { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Link, useSearchParams } from 'react-router-dom'
+import AppButton from '~/components/app-button/AppButton'
+import AsyncAutocomplete from '~/components/async-autocomlete/AsyncAutocomplete'
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import { styles } from '~/containers/subjects-search/SubjectsSearch.styles'
+import useBreakpoints from '~/hooks/use-breakpoints'
+import { authRoutes } from '~/router/constants/authRoutes'
+import { categoryService } from '~/services/category-service'
+
+const SubjectsSearch = () => {
+  const { t } = useTranslation()
+  const { isMobile } = useBreakpoints()
+  const [searchParams, setSearchParams] = useSearchParams()
+  const [subjectName, setSubjectName] = useState(
+    searchParams.get('subjectName') || ''
+  )
```

Comment:
https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state
```suggestion
  const [subjectName, setSubjectName] = useState(
    ()=>(searchParams.get('subjectName') || '')
  )
```